src/material-experimental/mdc-button/harness/button-harness.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ButtonHarnessFilters} from './button-harness-filters';
+
+
+/**
+ * Harness for interacting with a standard mat-button in tests.
+ * @dynamic
+ */
+export class MatButtonHarness extends ComponentHarness {
+  static hostSelector = [
+    '[mat-button]',
+    '[mat-raised-button]',
+    '[mat-flat-button]',
+    '[mat-icon-button]',
+    '[mat-stroked-button]',
+    '[mat-fab]',
+    '[mat-mini-fab]',
+  ].join(',');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a button with specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `label` finds a button with specific label text.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ButtonHarnessFilters = {}): HarnessPredicate<MatButtonHarness> {
+    return new HarnessPredicate(MatButtonHarness)
+        .addOption('text', options.text,
+            (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text));
+  }
+
+  /** Clicks the button. */
+  async click(): Promise<void> {
+    return (await this.host()).click();
+  }
+
+  /** Gets a boolean promise indicating if the button is disabled. */
+  async isDisabled(): Promise<boolean> {
+    const disabled = (await this.host()).getAttribute('disabled');
+    return coerceBooleanProperty(await disabled);
+  }
+
+  /** Gets a promise for the button's name. */
+  async getName(): Promise<string | null> {
```

Comment:
Is getting the name of a button common enough for it to have it's own method? We could expose a `getAttribute` instead which would cover more cases.

Reply:
Removed. I was on the fence myself, having just left it in there from checkbox